cjworkbench/i18n/trans.py

```diff
@@ -42,14 +44,35 @@ def _get_translations(locale):
     return _translators[locale]
 
 
-def trans(locale, message_id, *, default, context=None, parameters={}, tags={}):
+def trans(message_id, *, default, context=None, parameters={}, tags={}):
```

Comment:
It looks like we're making one function do two things here. Please refactor so we have a total of three functions:

1. `trans()` -- no `context`, no `tags`; outputs normal text. May output unescaped HTML special characters (we escape them elsewhere). Maybe we can even survive without `parameters` support -- meaning, we could maybe wrap Babel's own `trans()` directly. (I'm out of my element here; you may correct me....)
2. `trans_lazy()` -- like `trans()`
3. `trans_html()` -- for our `{% trans %}` template tag -- returns `html_safe()` text. This is the function we worked so hard on -- and it should only be called directly from HTML templates.

My point: `trans_html()` is exclusively for HTML templates; `trans()` is not. They shouldn't be the same function because their users should have different expectations about how they work.

Reply:
OK, splitting them seems a very good idea.

Babel is written with `gettext` in mind, so we don't want to use it for actually translating stuff.

Regarding `context`: It should also be in `trans()`, since it has to do with message catalogs.

Reply:
k cool. I was confusing it with Django template-tag context.